lib/internal/streams/async_iterator.js

```diff
@@ -182,6 +191,8 @@ const createReadableStreamAsyncIterator = (stream) => {
 
   stream.on('readable', onReadable.bind(null, iterator));
 
+  stream[kIterator] = iterator;
```

Comment:
Would be preferable to do like `weakSetOfStreams.add(stream)` so we don't ref `iterator`.

Reply:
yes, I did it like that for the potential re-use case.